### PR TITLE
Thrift zero-copy optimization for TElasticFramedTransport

### PR DESCRIPTION
**Pull request by MrQuansy**
Previously, these methods of TElasticFramedTransport are not override implemented.
TBinaryProtocol will allocate new byte array and copy from frame buffer. After this pr, TBinaryProtocol can wrap the byte buffer with the reference of frame buffer segment, which eliminates the cost of memory allocation and memory copying. 

image

We use benchmark to evaluate this optimization (Insert Tablet with text value type):
* Before
image
* After
image

### CODE REVIEW

**Comment by MrQuansy** [comment]
@OneSizeFitsQuorum PTAL

**Review by JackieTien97 — APPROVED** [review]

File: iotdb-client/service-rpc/src/main/java/org/apache/iotdb/rpc/RpcTransportFactory.java [hunk 1]
```diff
@@ -19,105 +19,28 @@
...
-import org.apache.thrift.transport.TMemoryInputTransport;
-import org.apache.thrift.transport.TSSLTransportFactory;
-import org.apache.thrift.transport.TSocket;
-import org.apache.thrift.transport.TTransport;
-import org.apache.thrift.transport.TTransportException;
 import org.apache.thrift.transport.TTransportFactory;
 
-@SuppressWarnings("java:S1135") // ignore todos
-public class RpcTransportFactory extends TTransportFactory {
+public class RpcTransportFactory extends BaseRpcTransportFactory {
```
Comment (OneSizeFitsQuorum) [inline iotdb-client/service-rpc/src/main/java/org/apache/iotdb/rpc/RpcTransportFactory.java:24]: ZeroCopyRpcTransportFactory

File: iotdb-client/service-rpc/src/main/java/org/apache/iotdb/rpc/TimeoutChangeableTFastFramedTransport.java [hunk 2]
```diff
@@ -56,19 +56,22 @@ public static class Factory extends TTransportFactory {
...
 
     @Override
     public TTransport getTransport(TTransport trans) {
       if (trans instanceof TSocket) {
         return new TimeoutChangeableTFastFramedTransport(
-            (TSocket) trans, thriftDefaultBufferSize, thriftMaxFrameSize);
+            (TSocket) trans, thriftDefaultBufferSize, thriftMaxFrameSize, copyBinary);
       } else {
-        return new TElasticFramedTransport(trans, thriftDefaultBufferSize, thriftMaxFrameSize);
+        return new TElasticFramedTransport(
```
Comment (OneSizeFitsQuorum) [inline iotdb-client/service-rpc/src/main/java/org/apache/iotdb/rpc/TimeoutChangeableTFastFramedTransport.java:73]: copyBinary

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Pull request merged**